components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -16,19 +11,46 @@
 #include "TimeLevel.hpp"
 #include "Tracers.hpp"
 #include "Types.hpp"
-#include "mpi/MpiContext.hpp"
+
+// Scream includes
+#include "dynamics/homme/homme_dimensions.hpp"
+#include "dynamics/homme/homme_dynamics_helpers.hpp"
+#include "dynamics/homme/homme_inputs_initializer.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "dynamics/homme/interface/scream_homme_interface.hpp"
 
 namespace scream
 {
 
-HommeDynamics::HommeDynamics (const ekat::Comm& comm, const ekat::ParameterList& /* params */)
- : m_dynamics_comm (comm)
+HommeDynamics::HommeDynamics (const ekat::Comm& comm, const ekat::ParameterList& params)
+ : m_params        (params)
+ , m_dynamics_comm (comm)
 {
-  init_homme1(comm);
+  // Init homme context par struct
+  if (!is_parallel_inited_f90()) {
+    auto comm_f = MPI_Comm_c2f(comm.mpi_comm());
+    init_parallel_f90(comm_f);
+  }
+
+  // Load homme params from namelist
+  if (!is_params_inited_f90()) {
+    auto nl_fname = m_params.get<std::string>("Homme namelist filename","namelist.nl");
```

Comment:
Should the default namelist be "user_nl_cam"?

Reply:
I used the name that homme uses as default in its internal tests. If you prefer, we could even remove the default (causing a runtime error if the nl name is not set in the input yaml file...

Reply:
I would vote for deleting the default and throwing an error.  A physics developer won't know about the homme internal tests and may get the impression that a namelist.nl file is required.  If we decide to include a default namelist with the codebase we can always add the default name back in here.  Thoughts?